stack-example/README.md

```diff
@@ -0,0 +1,83 @@
+This setup shows how to index haskell packages using `stack`
+and the [haskell-indexer-plugin][haskell-indexer-plugin].
+
+[haskell-indexer-plugin]: https://github.com/google/haskell-indexer
+
+First change directory to `stack-example` and build the plugin.
+```bash
+$ cd stack-example
+$ stack build haskell-indexer-plugin
+```
+
+Then generate a ghc wrapper script. This script will pass
+to GHC all the flags needed to enable the indexer plugin.
+```bash
+$ ./gen-ghc-wrapper.sh
+
+USAGE
+
+    ./gen-ghc-wrapper.sh OUTDIR GHC_PATH PLUGIN_DB
+
+Generates a wrapper for a ghc compiler in OUTDIR that indexes
+using the given compiler.
+
+EXAMPLE
+
+    ./gen-ghc-wrapper.sh ghc_wrapper "$(stack path --compiler-exe)" \
+    "$(stack path --snapshot-pkg-db)"
```

Comment:
Note: if not using the locally provided stack.yaml that seems to place these packages into the snapshot, then after `stack install`-ing the `haskell-indexer-plugin`, I could also pass `stack path --local-pkg-db` so the plugin were found.

Could you describe the benefit of using the snapshot? It probably has some use-case, but I'm not a hard-core stack user. Would be nice to add to top of the provided stack.yaml for example.

Also would be nice to call attention to override/undef `STACK_YAML` if any, I was puzzled a bit until I figured I have it hanging over from previous operations.

Reply:
I have no idea about this @facundominguez added all the stuff about stack.

Reply:
> I could also pass stack path --local-pkg-db so the plugin were found.

This would point to the database containing the plugin, but all the snapshot dependencies of the plugin would be missing because they still would be installed in the snapshot database.

Using a snapshot puts everything in a single database.